### DISCUSSION TITLE
OPA HPA

### DISCUSSION THREAD

**Discussion by rh-at** [apache/skywalking#11495] [General]
Hi there,
can i scale the OPA deployment itself using an HPA from with some sort of metrics that comes from the adapter?
and if so, what kind of metrics does SW recommends for that ?

**Comment by wu-sheng** [comment]
We don't have suggestions, and it is rare to scale OAP because this should be a predictable cost.
Anyway, if you want to, all metrics on the self-observability page are available for you.

You could use PromQL to query those or some tools through GraphQL to trigger the HPA.
